Sources/Jetworking/Client.swift

```diff
@@ -0,0 +1,152 @@
+        let dataTask = session.dataTask(with: request) { [weak self] data, urlResponse, error in
+            self?.handleResponse(data: data, urlResponse: urlResponse, error: error, endpoint: endpoint, completion: completion)
+        }
+
+        dataTask.resume()
+    }
+
+    private func createRequest(forHttpMethod httpMethod: HTTPMethod, andPathComponent pathComponent: String) -> URLRequest {
+        return .init(
+            url: clientConfiguration.baseURL.appendingPathComponent(pathComponent),
+            httpMethod: httpMethod,
+            headerFields: getHeaderFields()
+        )
+    }
+
+    private func getHeaderFields() -> [String: String] {
+        var headerFields: [String: String] = [
+            "Accept": "application/json",
+            "Content-Type": "application/json"
+        ]
+
+        if
+            let authorizationHeader: [String: String] = getAuthorizationHeader(),
+            let authorizationKey = authorizationHeader.keys.first,
+            let authorizationValue = authorizationHeader.values.first
+        {
+            headerFields[authorizationKey] = authorizationValue
+        }
+
+        return headerFields
+    }
+
+    private func getAuthorizationHeader() -> [String: String]? {
+        let authorizationHeaderKey: String = "Authorization"
+
+        switch clientConfiguration.authenticationMethod {
+        case .none:
+            return nil
+
+        case let .basicAuthentication(username, password):
+            var authString: String = ""
+            let credentialsString = "\(username):\(password)"
+            if let credentialsData = credentialsString.data(using: .utf8) {
+                let base64Credentials = credentialsData.base64EncodedString(options: [])
+                authString = "Basic \(base64Credentials)"
+            }
+
+            return [authorizationHeaderKey: authString]
+
+        case let .bearerToken(token):
+            return [authorizationHeaderKey: "Bearer \(token)"]
+
+        case let .custom(headerKey, headerValue):
+            return [headerKey: headerValue]
+        }
+    }
+
+    private func handleResponse<ResponseType>(
+        data: Data?,
+        urlResponse: URLResponse?,
+        error: Error?,
+        endpoint: Endpoint<ResponseType>,
+        completion: @escaping (Result<ResponseType, Error>) -> Void
+    ) {
+        if let error = error { return completion(.failure(error)) }
+
+        guard let urlResponse = urlResponse as? HTTPURLResponse else { return completion(.failure(APIError.responseMissing)) }
+
+        let statusCode = HTTPStatusCodeType(statusCode: urlResponse.statusCode)
+
+        switch statusCode {
+        case .successful:
+            guard
+                let data = data,
+                let decodedData = try? clientConfiguration.decoder.decode(ResponseType.self, from: data)
+            else { return completion(.failure(APIError.decodingError)) }
```

Comment:
I suggest to pass the underlying error of the decoding error, then it is possible to look into it which property couldn't decoded. Maybe as parameter of the decodingError case. But then we have to use `try` without optional.

Reply:
This is just a first version of the framework. I will add tickets so we create a proper error handling. Maybe this should also be discussed in a larger group of contributors.